### PR TITLE
[3305][15.0][MIG] product_carousel_image_attachment: Migration to 15.0

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
[3305](https://www.quartile.co/web#id=3305&action=771&model=project.task&view_type=form&menu_id=505)

### CODE REVIEW

File: product_carousel_image_attachment/models/__init__.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
```
Comment (yostashiro) [inline product_carousel_image_attachment/models/__init__.py]:
```suggestion
```

File: product_carousel_image_attachment/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+This module does when an image (.jpg/.png/.gif/.bmp) is attached to a product, create a
+corresponding product.image record for the attached file.
```
Comment (yostashiro) [inline product_carousel_image_attachment/readme/DESCRIPTION.rst]:
```suggestion
This module does the following:

- When an image (.jpg/.png/.gif/.bmp) is attached to a product, create a corresponding product.image record
  for the attached file.
```

**Review by yostashiro — CHANGES_REQUESTED** [review]
@AungKoKoLin1997 Can you please explain how image resizing is irrelevant now, since it was once of the key concerns not to allow uploading large image files to keep the data volume subdued.

**Comment by AungKoKoLin1997** [comment]
I thought image_resize_image() is not required anymore because product_template_image_ids and product_variant_image_ids use the image field type and they will resize themself. https://www.odoo.com/forum/help-1/which-function-odoo-13-replace-image-resize-image-164007

**Comment by yostashiro** [comment]
> I thought image_resize_image() is not required anymore because product_template_image_ids and product_variant_image_ids use the image field type and they will resize themself. https://www.odoo.com/forum/help-1/which-function-odoo-13-replace-image-resize-image-164007

@AungKoKoLin1997 If I'm not mistaken, the data volume will increase by x4 or so.  Can you please check and verify if this is the correct understanding?

**Comment by AungKoKoLin1997** [comment]
> If I'm not mistaken, the data volume will increase by x4 or so. Can you please check and verify if this is the correct understanding?

Yes, the volume will increased because when we add value to image_1920, system will resize the image and stored in relevant fields.

**Comment by yostashiro** [comment]
@AungKoKoLin1997 OK, we may wait and see how things will go with attachment_s3.  If x4 data is not much a concern (i.e. cost impact is neglegible) then we could abandon the resizing logic.

**Comment by kanda999** [comment]
@AungKoKoLin1997` 

I got that error when I deplicate res_user
` File "<decorator-gen-112>", line 2, in create
  File "/odoo/code/odoo/odoo/api.py", line 413, in _model_create_multi
    return create(self, arg)
  File "/odoo/code/odoo/odoo/addons/base/models/res_users.py", line 1386, in create
    users = super(UsersView, self).create(new_vals_list)
  File "<decorator-gen-110>", line 2, in create
  File "/odoo/code/odoo/odoo/api.py", line 413, in _model_create_multi
    return create(self, arg)
  File "/odoo/code/odoo/odoo/addons/base/models/res_users.py", line 1139, in create
    return super(UsersImplied, self).create(vals_list)
  File "<decorator-gen-104>", line 2, in create
  File "/odoo/code/odoo/odoo/api.py", line 413, in _model_create_multi
    return create(self, arg)
  File "/odoo/code/odoo/odoo/addons/base/models/res_users.py", line 561, in create
    users = super(Users, self).create(vals_list)
  File "<decorator-gen-67>", line 2, in create
  File "/odoo/code/odoo/odoo/api.py", line 413, in _model_create_multi
    return create(self, arg)
  File "/odoo/code/odoo/odoo/addons/base/models/ir_fields.py", line 613, in create
    recs = super().create(vals_list)
  File "<decorator-gen-13>", line 2, in create
  File "/odoo/code/odoo/odoo/api.py", line 413, in _model_create_multi
    return create(self, arg)
  File "/odoo/code/odoo/odoo/models.py", line 4071, in create
    parents = self.env[model_name].create([
  File "<decorator-gen-190>", line 2, in create
  File "/odoo/code/odoo/odoo/api.py", line 413, in _model_create_multi
    return create(self, arg)
  File "/odoo/code/odoo/addons/account/models/partner.py", line 553, in create
    return super().create(vals_list)
  File "<decorator-gen-98>", line 2, in create
  File "/odoo/code/odoo/odoo/api.py", line 413, in _model_create_multi
    return create(self, arg)
  File "/odoo/code/odoo/odoo/addons/base/models/res_partner.py", line 621, in create
    partners = super(Partner, self).create(vals_list)
  File "<decorator-gen-133>", line 2, in create
  File "/odoo/code/odoo/odoo/api.py", line 413, in _model_create_multi
    return create(self, arg)
  File "/odoo/code/odoo/addons/mail/models/mail_thread.py", line 265, in create
    threads = super(MailThread, self).create(vals_list)
  File "<decorator-gen-205>", line 2, in create
  File "/odoo/code/odoo/odoo/api.py", line 413, in _model_create_multi
    return create(self, arg)
  File "/odoo/code/odoo/addons/website/models/mixins.py", line 211, in create
    records = super(WebsitePublishedMixin, self).create(vals_list)
  File "<decorator-gen-67>", line 2, in create
  File "/odoo/code/odoo/odoo/api.py", line 413, in _model_create_multi
    return create(self, arg)
  File "/odoo/code/odoo/odoo/addons/base/models/ir_fields.py", line 613, in create
    recs = super().create(vals_list)
  File "<decorator-gen-13>", line 2, in create
  File "/odoo/code/odoo/odoo/api.py", line 413, in _model_create_multi
    return create(self, arg)
  File "/odoo/code/odoo/odoo/models.py", line 4079, in create
    records = self._create(data_list)
  File "/odoo/code/odoo/odoo/models.py", line 4229, in _create
    field.create([
  File "/odoo/code/odoo/odoo/fields.py", line 2289, in create
    super(Image, self).create(new_record_values)
  File "/odoo/code/odoo/odoo/fields.py", line 2193, in create
    env['ir.attachment'].sudo().with_context(
  File "<decorator-gen-248>", line 2, in create
  File "/odoo/code/odoo/odoo/api.py", line 413, in _model_create_multi
    return create(self, arg)
  File "/odoo/code/sst-custom/product_carousel_image_attachment/models/ir_attachment.py", line 37, in create
    return attachment
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/odoo/code/odoo/odoo/http.py", line 643, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/odoo/code/odoo/odoo/http.py", line 301, in _handle_exception
    raise exception.with_traceback(None) from new_cause
UnboundLocalError: local variable 'attachment' referenced before assignment`

**Comment by AungKoKoLin1997** [comment]
@kanda999  I fixed it. Thanks for the feedback.

**Review by rinaldifirdaus — APPROVED** [review]
I have done functional test for this module and it works well. and about the image size for zoom in the image, as discussed with customer that the image only can be zoomed in for image with 1025 x 1025 in this v15.

**Comment by kanda999** [comment]
This PR is used by rmm-custom.　https://github.com/qrtl/rmm-custom/pull/79

**Pull request closed without merging**